compiler/rustc_trait_selection/src/solve/assembly.rs

```diff
@@ -342,15 +340,15 @@ impl<'tcx> EvalCtxt<'_, 'tcx> {
             ty::Alias(_, alias_ty) => alias_ty,
         };
 
-        for (i, (assumption, _)) in self
+        for (_, (assumption, _)) in self
             .tcx()
             .bound_explicit_item_bounds(alias_ty.def_id)
             .subst_iter_copied(self.tcx(), alias_ty.substs)
             .enumerate()
```

Comment:
Here you can drop the `.enumerate()` in order to iterate over `(assumption, _)` without pairing the (no longer used) index to the iterator.